projects/opendr_ws_2/src/opendr_perception/README.md

```diff
@@ -0,0 +1,874 @@
+    ```
+    The following optional arguments are available:
+   - `-h or --help`: show a help message and exit
+   - `-i or --input_rgb_image_topic INPUT_RGB_IMAGE_TOPIC`: topic name for input RGB image (default=`/image_raw`)
+   - `-o or --output_rgb_image_topic OUTPUT_RGB_IMAGE_TOPIC`: topic name for output annotated RGB image, `None` to stop the node from publishing on this topic (default=`/opendr/image_emotion_estimation_annotated`)
+   - `-e or --output_emotions_topic OUTPUT_EMOTIONS_TOPIC`: topic to which we are publishing the facial emotion results, `None` to stop the node from publishing on this topic (default=`"/opendr/facial_emotion_estimation"`)
+   - `-m or --output_emotions_description_topic OUTPUT_EMOTIONS_DESCRIPTION_TOPIC`: topic to which we are publishing the description of the estimated facial emotion, `None` to stop the node from publishing on this topic (default=`/opendr/facial_emotion_estimation_description`)
+   - `--device DEVICE`: device to use, either `cpu` or `cuda`, falls back to `cpu` if GPU or CUDA is not found (default=`cuda`)
+
+3. Default output topics:
+   - Output images: `/opendr/image_emotion_estimation_annotated`
+   - Detection messages: `/opendr/facial_emotion_estimation`, `/opendr/facial_emotion_estimation_description`
+
+   For viewing the output, refer to the [notes above.](#notes)
+
+**Notes**
+
+This node requires the detection of a face first. This is achieved by including of the toolkit's face detector and running face detection on the input.
+Afterwards, the detected bounding box of the face is cropped and fed into the facial emotion estimator. 
+Feel free to modify the node to detect faces in a different way that matches your use case.
+
+### Landmark-based Facial Expression Recognition ROS2 Node
+
+A ROS2 node for performing landmark-based facial expression recognition using the pretrained model PST-BLN on AFEW, CK+ or Oulu-CASIA datasets.
+
+You can find the landmark-based facial expression recognition ROS2 node python script [here](./opendr_perception/landmark_based_facial_expression_recognition_node.py) to inspect the code and modify it as you wish to fit your needs.
+The node makes use of the toolkit's landmark-based facial expression recognition tool which can be found [here](../../../../src/opendr/perception/facial_expression_recognition/landmark_based_facial_expression_recognition/progressive_spatio_temporal_bln_learner.py)
+whose documentation can be found [here](../../../../docs/reference/landmark-based-facial-expression-recognition.md).
+
+#### Instructions for basic usage:
+
+1. Start the node responsible for publishing images. If you have a USB camera, then you can use the `usb_cam_node` as explained in the [prerequisites above](#prerequisites).
+
+2. You are then ready to start the landmark-based facial expression recognition node:
+
+    ```shell
+    ros2 run opendr_perception landmark_based_facial_expression_recognition_node.py
```

Comment:
Getting this error:
```
  File "/home/daniel/opendr/projects/opendr_ws_2/install/opendr_perception/lib/opendr_perception/landmark_based_facial_expression_recognition", line 33, in <module>
    sys.exit(load_entry_point('opendr-perception==1.0.0', 'console_scripts', 'landmark_based_facial_expression_recognition')())
  File "/home/daniel/opendr/projects/opendr_ws_2/install/opendr_perception/lib/python3.8/site-packages/opendr_perception/landmark_based_facial_expression_recognition_node.py", line 167, in main
    LandmarkFacialExpressionRecognitionNode(
  File "/home/daniel/opendr/projects/opendr_ws_2/install/opendr_perception/lib/python3.8/site-packages/opendr_perception/landmark_based_facial_expression_recognition_node.py", line 90, in __init__
    self.expression_classifier.load(model_saved_path, model)
  File "/home/daniel/opendr/src/opendr/perception/facial_expression_recognition/landmark_based_facial_expression_recognition/progressive_spatio_temporal_bln_learner.py", line 732, in load
    with open(os.path.join(path, model_name + ".json")) as metadata_file:
FileNotFoundError: [Errno 2] No such file or directory: './pretrained_models/pstbln_afew/pstbln_afew.json'

```

Reply:
This one is generally problematic as we don't provide a pretrained model. Something to do with licensing and permissions. 

There's not much we can do from the ROS side, i think. Maybe just include suggestions to use the new image-based tool or train custom models. 

I think that @negarhdr has answered this before in some meetings.

Reply:
I think you're right, in which case then yea, should be sufficient to mention it in the documentation for this ros2 node

Reply:
I added a relevant notice on the node's section for both ROS1 an 2.